### PR TITLE
[DeadCode] Skip array_map in class extends Exception on RemoveUnusedPrivateMethodRector

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by samsonasik** [comment]
Fixed 🎉

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules/DeadCode/Rector/ClassMethod/RemoveUnusedPrivateMethodRector.php [hunk 1]
```diff
@@ -186,36 +183,74 @@ private function isClassMethodCalledInLocalMethodCall(Class_ $class, string $cla
...
 
             // the method is used
-            if ($this->isName($methodCall->name, $classMethodName)) {
+            if ($this->isName($call->name, $classMethodName)) {
                 return true;
             }
         }
 
         return false;
     }
```
Comment (TomasVotruba) [inline rules/DeadCode/Rector/ClassMethod/RemoveUnusedPrivateMethodRector.php:167]: Could you extract this method to decoupled service? It seems we might re-use it.

Comment (samsonasik) [reply rules/DeadCode/Rector/ClassMethod/RemoveUnusedPrivateMethodRector.php:167]: implemented https://github.com/rectorphp/rector-src/pull/598/commits/0c1972bbd1fb28c9221f1f67301e9d49b5871954

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Thank you :clap:

**Pull request merged**